docs/package-management.md

```diff
@@ -0,0 +1,137 @@
+environments provide more stability, especially when being loaded automatically.
+For example, if you install one version of a package, load it with a `spack
+load` command in your `.bashrc` file, and later install another version of the
+same package, then the `spack load` command in your `.bashrc` can suddenly stop
+working. Environments have an internal lock file that articulates the exact
+packages that are loaded, so they're safe to load from a `.bashrc` file or in an
+automated script. When Spack is required for an interactive application, we use
+Spack environments for this reason.
+
+The Spack documentation provides a [tutorial on
+environments](https://spack-tutorial.readthedocs.io/en/latest/tutorial_environments.html),
+which is useful for a more comprehensive understanding of how these environments
+function. In this documentation, we'll take a more focused approach, dealing
+with common setup and installation tasks for preparing an environment for a
+course.
+
+You can only set up environments in environments that you can modify. In HUIT
+Open OnDemand, that means that you can only set up environments in your own
+personal Spack installation in your home folder, or in a shared course folder.
+In both cases, you should have the global spack installation set as an upstream,
+whether by the course shared folder's pre-made spack installation script, or by
+following our directions on setting up your own Spack installation.
+
+For the rest of this document, we'll assume that you've already sourced the
+`setup-env.sh` script from the installation where you want the environment to
+be.
+
+### Creating & Activating
+
+For an environment to be used, it must first be created. This is pretty
+straightforward, with the command
+
+```bash
+spack env create environmentName
+```
+
+Replace "environmentName" with whatever you want the environment to be called.
+Then the environment can be activated with
+
+```bash
+spack env activate environmentName
+```
+
+### Adding Packages
+
+To get started, you can add one or more packages. You do this with a command like
+
+```bash
+spack add packageSpec
+```
+
+You can install a package simply by name, but if you have specific version or
+compiler requirements, you can specify those with [Spack's spec
+syntax](https://spack.readthedocs.io/en/latest/basic_usage.html#sec-specs).
+
+Adding a package won't install it, but it will associate that package with the
+environment itself.
+
+### Installing Packages
+
+The syntax to start an installation is simple, but where to run the installation
+is less simple. It's best to run these installations in a Slurm job, since some
+installations will take a long time to run. Therefore, we recommend adapting
+this bash script for your Spack installation, and using it via `sbatch`. This is
+the script that we use to manage the global installation, so just change the
+line with the Spack activation to match your installation path.
+
```

Comment:
I wonder if we should explicitly say create a bash script and copy these contents into that script and change these variables. For example: `You will need to create a bash script/file and copy the code below into the script/file. In addition you will need to change the path(line commented with #CHANGEME) so the script can enable spack`